docs/quantum_oscilator/architecture.md

```diff
@@ -0,0 +1,81 @@
+# Architecture of Neutal Network
+
+## Parent classes
+
+[`#!python class tf.keras.Model`](https://keras.io/api/models/model/)
+
+## Introduction
+
+QONetwork class is responsible for the learning process from the user input.
+Our neural network architecture that simultaneously learns eigenvalues and the
+associated eigenfunctions using a scanning mechanism. The proposed technique is
+an extension to neural network differential equation solvers and, consequently,
+acquires all the benefits that network solvers have over numerical integrators.
+
+### Network structure
+
+| name                                                                                                | type               | shape |
+| --------------------------------------------------------------------------------------------------- | ------------------ | ----- |
+| [`input_1 ` `(InputLayer) `](https://www.tensorflow.org/api_docs/python/tf/keras/layers/InputLayer) | `#!python float32` | `1`   |
+| `eigenvalue ` `(Eigenvalue)`                                                                        | `#!python float32` | `1`   |
+| [`concatenate ` `(Cocatenate)`](https://keras.io/api/layers/merging_layers/concatenate/)            | `#!python float32` | `2`   |
+| [`dense_1 ` `(Dense)`](https://keras.io/api/layers/core_layers/dense/)                              | `#!python float32` | `2`   |
+| [`dense_2 ` `(Dense)`](https://keras.io/api/layers/core_layers/dense/)                              | `#!python float32` | `50`  |
+| [`predictions ` `(Dense)`](https://keras.io/api/layers/core_layers/dense/)                          | `#!python float32` | `1`   |
+
+### Abstract
+
+The construction of the network involves the definition of layers in the
+assemble_hook method. In neural networks there is one input layer (input), one
+output layer (output) and many hidden layers (d1, d2, ...). Adding hidden
+layers involves adding consecutive lines of code in the form dX =
+keras.layers.Dense(number_neurons, activation function, layer_name, number
+precision)(layer_previous).
+
+When building a neural network architecture, parameters within each layer can
+be manipulated, e.g.: different activation functions (sin, relu, tanh), number
+of neurons in each hidden layer, and changing the precision of the data.
+
+## Instance attributes
+
+| name            | description                             |
+| --------------- | --------------------------------------- |
+| constants       | constant values used in neutral network |
+| is_debug        |                                         |
+| loss_function   | loss function of neutral network        |
+| is_console_mode |                                         |
+
+## Constructor
+
```

Comment:
Change contructor to function signature and include parameters and returns table similar to how QOParams methods were described

Reply:
1. Co to sygnatura funkcji?
2. Gdzie ty masz te tabelki w QQParams?????????

Reply:
1. https://developer.mozilla.org/en-US/docs/Glossary/Signature/Function

2. My bad, what i ment is this thing:
```
### `#!python def push_stats(self, ...) -> None`

Called after each learning epoch of the network. Adds new values to the history
of learning process metrics values.

#### Parameters

Each parameter will be added to the corresponding class attribute (see names).

| name              | type             |
| ----------------- | ---------------- |
| `total_loss`      | `#!python float` |
| `eigenvalue`      | `#!python float` |
| `residuum`        | `#!python float` |
| `function_loss`   | `#!python float` |
| `eigenvalue_loss` | `#!python float` |
| `drive_loss`      | `#!python float` |
| `c`               | `#!python float` |

#### Returns

| type   | description                    |
| ------ | ------------------------------ |
| Figure | Figure used for drawing plots. | 

```